lib/runtime/procs/read.js

```diff
@@ -50,9 +34,25 @@ var Read = source.extend({
 
         this.validate_options(adapter.read.allowedOptions(), adapter.read.requiredOptions());
 
-        var defaultTimeRange = this.adapter.defaultTimeRange();
-        this.from = this.options.from || defaultTimeRange.from;
-        this.to = this.options.to || defaultTimeRange.to;
+        var defaultTimeOptions = this.adapter.defaultTimeOptions();
+        this.from = this.options.from || defaultTimeOptions.from;
+        this.to = this.options.to || defaultTimeOptions.to;
+        this.lag = options.lag || defaultTimeOptions.lag || JuttleMoment.duration(0, 's');
+        this.readEvery = options.every || defaultTimeOptions.every || JuttleMoment.duration(1, 's');
+
+        if (! this.lag.duration) {
```

Comment:
I think these type checks should go into `handleTimeOptions()` so that they execute before initializing the adapter. That way if the adapter for some reason wants to verify these options they can be sure it's the right type.